oceannavigator/frontend/src/components/Map.jsx

```diff
@@ -1260,17 +1266,18 @@ export default class Map extends React.PureComponent {
       // Disable zooming when drawing
       this.controlDoubleClickZoom(false);
       const lonlat = olproj.transform(e.feature.getGeometry().getCoordinates(), this.props.state.projection, "EPSG:4326");
```

Comment:
I think can avoid all of the messy `[lonlat[1], lonlat[0]]` array allocations by simply reversing this array. Note you'll have to change your variable names from `lonlat` to `latlon` where appropriate.

```js
const latlon = olproj
    .transform(e.feature.getGeometry().getCoordinates(), this.props.state.projection, "EPSG:4326")
    .reverse();

const drawn_latlons = [...this.state.latlon, latlon];
```